Mathlib/Algebra/MonoidAlgebra/Ideal.lean

```diff
@@ -28,31 +28,34 @@ theorem MonoidAlgebra.mem_ideal_span_of_image [Monoid G] [Semiring k] {s : Set G
     x ∈ Ideal.span (MonoidAlgebra.of k G '' s) ↔ ∀ m ∈ x.support, ∃ m' ∈ s, ∃ d, m = d * m' := by
   let RHS : Ideal (MonoidAlgebra k G) :=
     { carrier := { p | ∀ m : G, m ∈ p.support → ∃ m' ∈ s, ∃ d, m = d * m' }
-      add_mem' := fun x y hx hy m hm => by
+      add_mem' := @fun x y hx hy m hm => by
         classical exact (Finset.mem_union.1 <| Finsupp.support_add hm).elim (hx m) (hy m)
       zero_mem' := fun m hm => by cases hm
-      smul_mem' := fun x y hy m hm =>
-        by
-        replace hm := finset.mem_bUnion.mp (Finsupp.support_sum hm)
-        obtain ⟨xm, hxm, hm⟩ := hm
-        replace hm := finset.mem_bUnion.mp (Finsupp.support_sum hm)
+      smul_mem' := fun x y hy m hm => by
+        classical
+        replace hm := Finset.mem_bunionᵢ.mp (Finsupp.support_sum hm)
+        obtain ⟨xm, -, hm⟩ := hm
+        replace hm := Finset.mem_bunionᵢ.mp (Finsupp.support_sum hm)
         obtain ⟨ym, hym, hm⟩ := hm
-        replace hm := finset.mem_singleton.mp (Finsupp.support_single_subset hm)
+        replace hm := Finset.mem_singleton.mp (Finsupp.support_single_subset hm)
         obtain rfl := hm
-        refine' (hy _ hym).imp fun sm => Exists.imp fun hsm => _
+        -- porting note: changed `Exists.imp` to `And.imp_right` due to change in `∃ x ∈ s`
+        -- elaboration
+        refine' (hy _ hym).imp fun sm p => And.imp_right _ p
         rintro ⟨d, rfl⟩
         exact ⟨xm * d, (mul_assoc _ _ _).symm⟩ }
   change _ ↔ x ∈ RHS
   constructor
   · revert x
-    refine' Ideal.span_le.2 _
+    rw [←SetLike.le_def] -- porting note: refine needs this even though it's defeq?
+    refine Ideal.span_le.2 ?_
     rintro _ ⟨i, hi, rfl⟩ m hm
     refine' ⟨_, hi, 1, _⟩
-    obtain rfl := finset.mem_singleton.mp (Finsupp.support_single_subset hm)
+    obtain rfl := Finset.mem_singleton.mp (Finsupp.support_single_subset hm)
     exact (one_mul _).symm
   · intro hx
     rw [← Finsupp.sum_single x]
-    apply Ideal.sum_mem _ fun i hi => _
+    refine Ideal.sum_mem _ fun i hi => ?_  -- porting note: changed `apply` to `refine`
```

Comment:
For reviewers: this failure is really wierd, Lean totally ignores the `fun` and doesn't actually introduce `i` or `hi` if I use `apply`